stdlib/public/SDK/Foundation/NSStringAPI.swift

```diff
@@ -417,6 +417,20 @@ extension StringProtocol where Index == String.Index {
     return self._ephemeralString as NSString
   }
 
+  func _withCore<Result>(_ apply: (_StringCore) throws -> Result) rethrows -> Result {
```

Comment:
I wonder whether marking these `_withXXX` functions with `@_transparent` pr `@inline(__always)` will get us back the performance improvement. I heard, generic closures are very hard for the optimizer too look through.